// @hasaan_6/rag-chatbot-widget/src/index.ts
import "./styles/chatbot.css";

export { default as ChatbotWidget } from "./components/ChatbotWidget";
export { default as ChatWindow } from "./components/ChatWindow";
export { default as MessageBubble } from "./components/MessageBubble";
export { default as QuickActions } from "./components/QuickActions";
export { default as LeadForm } from "./components/LeadForm";

export type {
	ChatbotWidgetProps,
	ChatMessage,
	ChatSource,
	LeadFormValues,
	LeadFormConfig,
} from "./components/ChatbotWidget";

export type {
	EmbeddingAdapter,
	LLMAdapter,
	LLMAdapterInput,
	LLMAdapterOutput,
	ConversationTurn,
} from "./lib/adapters";

export { runRagPipeline } from "./lib/ragPipeline";
export type { RagPipelineConfig, RagPipelineResult, DocumentChunk } from "./lib/ragPipeline";

export {
	createGeminiEmbeddingAdapter,
	createOpenAIEmbeddingAdapter,
	createCohereEmbeddingAdapter,
	createOpenAICompatibleLLMAdapter,
	createDeepSeekLLMAdapter,
	createOpenAILLMAdapter,
	createGroqLLMAdapter,
	createMistralLLMAdapter,
	createTogetherLLMAdapter,
	createOllamaLLMAdapter,
	createAnthropicLLMAdapter,
	createGeminiLLMAdapter,
} from "./lib/adapterFactories";

export { ingestDocument, SUPABASE_SETUP_SQL } from "./lib/supabaseSetup";
