// @hasaan_6/rag-chatbot-widget/src/lib/adapters.ts
/**
 * Provider-agnostic adapter contracts for the RAG pipeline.
 *
 * Users implement these functions with any provider they want.
 */

// ─── Embedding Adapter ───────────────────────────────────────────────────────

/**
 * Implement this interface to connect any embedding provider.
 * The function receives the text to embed and must return a number array (the vector).
 *
 * @example Gemini (gemini-embedding-001, 3072 dimensions by default)
 * const embeddingAdapter: EmbeddingAdapter = async (text) => {
 *   const res = await fetch(
 *     `https://generativelanguage.googleapis.com/v1beta/models/gemini-embedding-001:embedContent?key=${GEMINI_KEY}`,
 *     { method: "POST", headers: { "Content-Type": "application/json" },
 *       body: JSON.stringify({ model: "models/gemini-embedding-001", content: { parts: [{ text }] } }) }
 *   );
 *   const data = await res.json();
 *   return data.embedding.values;
 * };
 *
 * @example OpenAI
 * const embeddingAdapter: EmbeddingAdapter = async (text) => {
 *   const res = await fetch("https://api.openai.com/v1/embeddings", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", Authorization: `Bearer ${OPENAI_KEY}` },
 *     body: JSON.stringify({ model: "text-embedding-3-small", input: text }),
 *   });
 *   const data = await res.json();
 *   return data.data[0].embedding;
 * };
 *
 * @example Cohere
 * const embeddingAdapter: EmbeddingAdapter = async (text) => {
 *   const res = await fetch("https://api.cohere.com/v2/embed", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", Authorization: `Bearer ${COHERE_KEY}` },
 *     body: JSON.stringify({ model: "embed-english-v3.0", texts: [text], input_type: "search_query", embedding_types: ["float"] }),
 *   });
 *   const data = await res.json();
 *   return data.embeddings.float[0];
 * };
 */
export type EmbeddingAdapter = (text: string) => Promise<number[]>;

// ─── LLM Adapter ─────────────────────────────────────────────────────────────

export type ConversationTurn = {
  role: "user" | "assistant";
  content: string;
};

export type LLMAdapterInput = {
  question: string;
  context: string;
  conversation: ConversationTurn[];
  systemPrompt: string;
};

export type LLMAdapterOutput = {
  answer: string;
  needsHumanHandoff?: boolean;
};

/**
 * Implement this interface to connect any LLM provider.
 * Receive the question, context chunks, and conversation history.
 * Return the generated answer string and optionally a handoff flag.
 *
 * @example DeepSeek (OpenAI-compatible)
 * const llmAdapter: LLMAdapter = async ({ systemPrompt, question, conversation }) => {
 *   const res = await fetch("https://api.deepseek.com/v1/chat/completions", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", Authorization: `Bearer ${DEEPSEEK_KEY}` },
 *     body: JSON.stringify({
 *       model: "deepseek-chat",
 *       messages: [{ role: "system", content: systemPrompt }, ...conversation, { role: "user", content: question }],
 *       temperature: 0.3, max_tokens: 1024,
 *     }),
 *   });
 *   const data = await res.json();
 *   return { answer: data.choices[0].message.content.trim() };
 * };
 *
 * @example OpenAI GPT-4o
 * const llmAdapter: LLMAdapter = async ({ systemPrompt, question, conversation }) => {
 *   const res = await fetch("https://api.openai.com/v1/chat/completions", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", Authorization: `Bearer ${OPENAI_KEY}` },
 *     body: JSON.stringify({
 *       model: "gpt-4o",
 *       messages: [{ role: "system", content: systemPrompt }, ...conversation, { role: "user", content: question }],
 *       temperature: 0.3,
 *     }),
 *   });
 *   const data = await res.json();
 *   return { answer: data.choices[0].message.content.trim() };
 * };
 *
 * @example Google Gemini (generateContent)
 * const llmAdapter: LLMAdapter = async ({ systemPrompt, question, conversation }) => {
 *   const contents = [
 *     ...conversation.map(t => ({ role: t.role === "assistant" ? "model" : "user", parts: [{ text: t.content }] })),
 *     { role: "user", parts: [{ text: question }] },
 *   ];
 *   const res = await fetch(
 *     `https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=${GEMINI_KEY}`,
 *     { method: "POST", headers: { "Content-Type": "application/json" },
 *       body: JSON.stringify({ system_instruction: { parts: [{ text: systemPrompt }] }, contents }) }
 *   );
 *   const data = await res.json();
 *   return { answer: data.candidates[0].content.parts[0].text.trim() };
 * };
 *
 * @example Anthropic Claude
 * const llmAdapter: LLMAdapter = async ({ systemPrompt, question, conversation }) => {
 *   const res = await fetch("https://api.anthropic.com/v1/messages", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", "x-api-key": CLAUDE_KEY, "anthropic-version": "2023-06-01" },
 *     body: JSON.stringify({
 *       model: "claude-3-5-haiku-20241022",
 *       max_tokens: 1024,
 *       system: systemPrompt,
 *       messages: [...conversation.map(t => ({ role: t.role, content: t.content })), { role: "user", content: question }],
 *     }),
 *   });
 *   const data = await res.json();
 *   return { answer: data.content[0].text.trim() };
 * };
 *
 * @example Mistral
 * const llmAdapter: LLMAdapter = async ({ systemPrompt, question, conversation }) => {
 *   const res = await fetch("https://api.mistral.ai/v1/chat/completions", {
 *     method: "POST",
 *     headers: { "Content-Type": "application/json", Authorization: `Bearer ${MISTRAL_KEY}` },
 *     body: JSON.stringify({
 *       model: "mistral-small-latest",
 *       messages: [{ role: "system", content: systemPrompt }, ...conversation, { role: "user", content: question }],
 *     }),
 *   });
 *   const data = await res.json();
 *   return { answer: data.choices[0].message.content.trim() };
 * };
 */
export type LLMAdapter = (input: LLMAdapterInput) => Promise<LLMAdapterOutput>;
