// @hasaan_6/rag-chatbot-widget/src/components/ChatbotWidget.tsx
"use client";

import { useEffect, useMemo, useState } from "react";
import { ChatWindow } from "./ChatWindow";
import { LeadForm, defaultLeadFormConfig, type LeadFormConfig } from "./LeadForm";
import { runRagPipeline } from "../lib/ragPipeline";
import type { ConversationTurn } from "../lib/adapters";

export type ChatRole = "user" | "assistant";

export type ChatSource = {
  title: string;
  url?: string | null;
  similarity?: number;
};

export type ChatMessage = {
  id: string;
  role: ChatRole;
  content: string;
  sources?: ChatSource[];
  needsHumanHandoff?: boolean;
};

export type { LeadFormConfig } from "./LeadForm";

export type ChatbotWidgetProps = {
  embeddingAdapter: import("../lib/adapters").EmbeddingAdapter;
  llmAdapter: import("../lib/adapters").LLMAdapter;
  supabaseUrl: string;
  supabaseAnonKey: string;
  matchCount?: number;
  matchThreshold?: number;
  conversationWindow?: number;
  botName?: string;
  botEyebrow?: string;
  toggleLabel?: string;
  inputPlaceholder?: string;
  welcomeMsg?: string;
  theme?: "light" | "dark";
  quickActions?: string[];
  fallbackMsg?: string;
  leadFormConfig?: LeadFormConfig;
  onLeadSubmit?: (values: LeadFormValues) => Promise<void>;
  leadEndpoint?: string;
};

export type LeadFormValues = {
  name: string;
  email: string;
  phone: string;
  companyName: string;
  requiredService: string;
  budget: string;
  projectDetails: string;
  preferredContactTime: string;
  sourcePage: string;
  honeypot: string;
  submittedAt: number;
};

const defaultWelcomeMsg =
  "Hi! I'm here to help. Ask me anything or pick one of the quick actions below.";

const defaultFallbackMsg =
  "I don't have enough information to answer that. Would you like to connect with our team?";

const defaultQuickActions = [
  "View Services",
  "View Case Studies",
  "Get a Quote",
  "Book a Free Consultation",
  "Contact Us",
  "Talk to Human Support",
];

const defaultBotName = "AI Assistant";
const defaultBotEyebrow = "Support";
const defaultToggleLabel = "Chat with us";
const defaultInputPlaceholder = "Type your message...";

function createInitialMessage(welcomeMsg: string): ChatMessage {
  return {
    id: crypto.randomUUID(),
    role: "assistant",
    content: welcomeMsg,
  };
}

function buildConversation(messages: ChatMessage[]): ConversationTurn[] {
  return messages.map((message) => ({
    role: message.role,
    content: message.content,
  }));
}

export default function ChatbotWidget({
  embeddingAdapter,
  llmAdapter,
  supabaseUrl,
  supabaseAnonKey,
  matchCount,
  matchThreshold,
  conversationWindow,
  botName = defaultBotName,
  botEyebrow = defaultBotEyebrow,
  toggleLabel = defaultToggleLabel,
  inputPlaceholder = defaultInputPlaceholder,
  welcomeMsg = defaultWelcomeMsg,
  theme = "light",
  quickActions = defaultQuickActions,
  fallbackMsg = defaultFallbackMsg,
  leadFormConfig,
  onLeadSubmit,
  leadEndpoint,
}: ChatbotWidgetProps) {
  const [open, setOpen] = useState(false);
  const [messages, setMessages] = useState<ChatMessage[]>(() => [createInitialMessage(welcomeMsg)]);
  const [input, setInput] = useState("");
  const [loading, setLoading] = useState(false);
  const [leadFormOpen, setLeadFormOpen] = useState(false);

  useEffect(() => {
    setMessages([createInitialMessage(welcomeMsg)]);
  }, [welcomeMsg]);

  const quickActionList = useMemo(() => quickActions.filter((action) => action.trim().length > 0), [quickActions]);
  const resolvedLeadFormConfig = useMemo(() => ({ ...defaultLeadFormConfig, ...leadFormConfig }), [leadFormConfig]);

  function getSourcePage() {
    return typeof window !== "undefined" ? window.location.pathname : "/";
  }

  async function submitQuestion(question: string) {
    const trimmedQuestion = question.trim();
    if (!trimmedQuestion || loading) {
      return;
    }

    const nextUserMessage: ChatMessage = {
      id: crypto.randomUUID(),
      role: "user",
      content: trimmedQuestion,
    };

    setMessages((current) => [...current, nextUserMessage]);
    setInput("");
    setLoading(true);

    try {
      const result = await runRagPipeline(trimmedQuestion, buildConversation([...messages, nextUserMessage]), {
        embeddingAdapter,
        llmAdapter,
        supabaseUrl,
        supabaseAnonKey,
        matchCount,
        matchThreshold,
        conversationWindow,
      });

      const assistantMessage: ChatMessage = {
        id: crypto.randomUUID(),
        role: "assistant",
        content: result.answer || fallbackMsg,
        needsHumanHandoff: result.needsHumanHandoff,
        sources: result.sources,
      };

      setMessages((current) => [...current, assistantMessage]);

      if (assistantMessage.needsHumanHandoff) {
        setLeadFormOpen(true);
      }
    } catch {
      setMessages((current) => [
        ...current,
        {
          id: crypto.randomUUID(),
          role: "assistant",
          content: fallbackMsg,
          needsHumanHandoff: true,
        },
      ]);
      setLeadFormOpen(true);
    } finally {
      setLoading(false);
    }
  }

  function handleQuickAction(action: string) {
    void submitQuestion(action);
  }

  async function submitLead(values: LeadFormValues) {
    if (leadEndpoint) {
      const response = await fetch(leadEndpoint, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(values),
      });

      if (!response.ok) {
        throw new Error("Unable to submit lead.");
      }
    } else if (onLeadSubmit) {
      await onLeadSubmit(values);
    }

    setLeadFormOpen(false);
    setMessages((current) => [
      ...current,
      {
        id: crypto.randomUUID(),
        role: "assistant",
        content: "Thanks! Your details have been received. Someone will be in touch with you shortly.",
      },
    ]);
  }

  const leadForm = leadFormOpen ? (
    <LeadForm
      sourcePage={getSourcePage()}
      config={resolvedLeadFormConfig}
      onCancel={() => setLeadFormOpen(false)}
      onSubmit={submitLead}
    />
  ) : null;

  return (
    <div className={`chatbot-widget-shell chatbot-theme-${theme}`}>
      <button
        type="button"
        onClick={() => setOpen((current) => !current)}
        className="chatbot-toggle"
        aria-label={open ? "Close chat" : "Open chat"}
        aria-expanded={open}
      >
        <span className="chatbot-toggle-icon" aria-hidden="true">
          <span className="chatbot-toggle-dot" />
        </span>
        <span className="chatbot-toggle-label">{toggleLabel}</span>
        <span className="chatbot-toggle-spark" aria-hidden="true">✦</span>
      </button>

      <ChatWindow
        open={open}
        messages={messages}
        quickActions={quickActionList}
        input={input}
        botName={botName}
        botEyebrow={botEyebrow}
        inputPlaceholder={inputPlaceholder}
        loading={loading}
        onClose={() => setOpen(false)}
        onInputChange={setInput}
        onSubmit={(event) => {
          event.preventDefault();
          void submitQuestion(input);
        }}
        onQuickAction={handleQuickAction}
      >
        {leadForm}
      </ChatWindow>
    </div>
  );
}
