// @hasaan_6/rag-chatbot-widget/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  ChatWindow: () => ChatWindow_default,
  ChatbotWidget: () => ChatbotWidget,
  LeadForm: () => LeadForm_default,
  MessageBubble: () => MessageBubble_default,
  QuickActions: () => QuickActions_default,
  SUPABASE_SETUP_SQL: () => SUPABASE_SETUP_SQL,
  createAnthropicLLMAdapter: () => createAnthropicLLMAdapter,
  createCohereEmbeddingAdapter: () => createCohereEmbeddingAdapter,
  createDeepSeekLLMAdapter: () => createDeepSeekLLMAdapter,
  createGeminiEmbeddingAdapter: () => createGeminiEmbeddingAdapter,
  createGeminiLLMAdapter: () => createGeminiLLMAdapter,
  createGroqLLMAdapter: () => createGroqLLMAdapter,
  createMistralLLMAdapter: () => createMistralLLMAdapter,
  createOllamaLLMAdapter: () => createOllamaLLMAdapter,
  createOpenAICompatibleLLMAdapter: () => createOpenAICompatibleLLMAdapter,
  createOpenAIEmbeddingAdapter: () => createOpenAIEmbeddingAdapter,
  createOpenAILLMAdapter: () => createOpenAILLMAdapter,
  createTogetherLLMAdapter: () => createTogetherLLMAdapter,
  ingestDocument: () => ingestDocument,
  runRagPipeline: () => runRagPipeline
});
module.exports = __toCommonJS(index_exports);

// #style-inject:#style-inject
function styleInject(css, { insertAt } = {}) {
  if (!css || typeof document === "undefined") return;
  const head = document.head || document.getElementsByTagName("head")[0];
  const style = document.createElement("style");
  style.type = "text/css";
  if (insertAt === "top") {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }
  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

// src/styles/chatbot.css
styleInject(":root {\n  --chatbot-primary: #6366f1;\n  --chatbot-primary-strong: #4f46e5;\n  --chatbot-secondary: #111827;\n  --chatbot-bg: #ffffff;\n  --chatbot-surface: #f9fafb;\n  --chatbot-text: #111827;\n  --chatbot-muted: rgba(17, 24, 39, 0.62);\n  --chatbot-border: rgba(17, 24, 39, 0.12);\n  --chatbot-danger: #b91c1c;\n  --chatbot-danger-bg: #fef2f2;\n  --chatbot-border-radius: 12px;\n}\n.chatbot-widget-shell {\n  color: var(--chatbot-text);\n}\n.chatbot-widget-shell,\n.chatbot-widget-shell * {\n  box-sizing: border-box;\n}\n.chatbot-theme-light {\n  --chatbot-bg: #ffffff;\n  --chatbot-surface: #f9fafb;\n  --chatbot-text: #111827;\n  --chatbot-muted: rgba(17, 24, 39, 0.62);\n  --chatbot-border: rgba(17, 24, 39, 0.12);\n}\n.chatbot-theme-dark {\n  --chatbot-bg: #0f172a;\n  --chatbot-surface: #111827;\n  --chatbot-text: #e5e7eb;\n  --chatbot-muted: rgba(229, 231, 235, 0.68);\n  --chatbot-border: rgba(255, 255, 255, 0.12);\n}\n.chatbot-toggle {\n  position: fixed;\n  right: 1.5rem;\n  bottom: 1.5rem;\n  z-index: 9999;\n  display: inline-flex;\n  align-items: center;\n  gap: 0.5rem;\n  border: 0;\n  border-radius: 9999px;\n  background: var(--chatbot-primary);\n  color: #fff;\n  padding: 0.75rem 1rem;\n  font-size: 0.875rem;\n  font-weight: 700;\n  line-height: 1;\n  box-shadow: 0 20px 45px rgba(15, 23, 42, 0.32);\n  transition:\n    transform 0.2s ease,\n    box-shadow 0.2s ease,\n    background-color 0.2s ease;\n}\n.chatbot-toggle:hover {\n  transform: translateY(-2px);\n  background: var(--chatbot-primary-strong);\n  box-shadow: 0 24px 52px rgba(15, 23, 42, 0.38);\n}\n.chatbot-toggle-icon {\n  display: grid;\n  width: 2.25rem;\n  height: 2.25rem;\n  place-content: center;\n  border-radius: 9999px;\n  background: rgba(255, 255, 255, 0.14);\n}\n.chatbot-toggle-dot {\n  width: 0.85rem;\n  height: 0.85rem;\n  border-radius: 9999px;\n  background: #fff;\n}\n.chatbot-toggle-spark {\n  color: rgba(255, 255, 255, 0.92);\n  font-size: 0.95rem;\n}\n.chatbot-window {\n  position: fixed;\n  right: 1.5rem;\n  bottom: 6rem;\n  z-index: 9998;\n  display: flex;\n  flex-direction: column;\n  width: min(calc(100vw - 1.5rem), 24rem);\n  max-height: calc(100vh - 7.5rem);\n  overflow: hidden;\n  border: 1px solid var(--chatbot-border);\n  border-radius: 28px;\n  background: var(--chatbot-bg);\n  box-shadow: 0 24px 80px rgba(15, 23, 42, 0.22);\n  transition: opacity 0.25s ease, transform 0.25s ease;\n}\n.chatbot-window-open {\n  opacity: 1;\n  pointer-events: auto;\n  transform: translateY(0);\n}\n.chatbot-window-closed {\n  opacity: 0;\n  pointer-events: none;\n  transform: translateY(1.5rem);\n}\n.chatbot-header {\n  position: relative;\n  flex: 0 0 auto;\n  overflow: hidden;\n  background:\n    linear-gradient(\n      135deg,\n      var(--chatbot-secondary),\n      var(--chatbot-primary));\n  color: #fff;\n  padding: 1rem 1.25rem;\n}\n.chatbot-header-glow {\n  position: absolute;\n  inset: 0;\n  background:\n    radial-gradient(\n      circle at top right,\n      rgba(255, 255, 255, 0.16),\n      transparent 40%),\n    radial-gradient(\n      circle at left,\n      rgba(255, 255, 255, 0.08),\n      transparent 35%);\n}\n.chatbot-header-content {\n  position: relative;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  gap: 1rem;\n}\n.chatbot-eyebrow,\n.chatbot-lead-kicker,\n.chatbot-field-label {\n  margin: 0;\n  font-size: 0.75rem;\n  font-weight: 700;\n  letter-spacing: 0.12em;\n  text-transform: uppercase;\n}\n.chatbot-eyebrow {\n  color: rgba(255, 255, 255, 0.7);\n}\n.chatbot-heading {\n  margin: 0;\n  color: #fff;\n  font-size: 1.125rem;\n  font-weight: 700;\n  line-height: 1.35;\n}\n.chatbot-close,\n.chatbot-link-button {\n  border: 0;\n  background: transparent;\n  cursor: pointer;\n}\n.chatbot-close {\n  flex-shrink: 0;\n  border: 1px solid rgba(255, 255, 255, 0.15);\n  border-radius: 9999px;\n  background: rgba(255, 255, 255, 0.1);\n  color: #fff;\n  padding: 0.4rem 0.75rem;\n  font-size: 0.875rem;\n  transition: background-color 0.2s ease;\n}\n.chatbot-close:hover {\n  background: rgba(255, 255, 255, 0.18);\n}\n.chatbot-body {\n  display: flex;\n  min-height: 0;\n  flex: 1 1 auto;\n  flex-direction: column;\n  gap: 1rem;\n  overflow-y: auto;\n  padding: 1rem;\n  background:\n    linear-gradient(\n      180deg,\n      rgba(99, 102, 241, 0.03),\n      transparent 24%);\n}\n.chatbot-form {\n  flex: 0 0 auto;\n  border-top: 1px solid var(--chatbot-border);\n  background: var(--chatbot-surface);\n  padding: 0.9rem 1rem 1rem;\n}\n.chatbot-input-shell {\n  display: flex;\n  align-items: center;\n  gap: 0.5rem;\n  min-height: 3.5rem;\n  border: 1px solid var(--chatbot-border);\n  border-radius: 9999px;\n  background: var(--chatbot-bg);\n  padding: 0.45rem 0.45rem 0.45rem 1rem;\n  box-shadow: 0 1px 2px rgba(15, 23, 42, 0.06);\n}\n.chatbot-input-shell:focus-within {\n  border-color: color-mix(in srgb, var(--chatbot-primary) 30%, var(--chatbot-border));\n  box-shadow: 0 0 0 4px rgba(99, 102, 241, 0.1);\n}\n.chatbot-input {\n  appearance: none;\n  box-shadow: none;\n  min-width: 0;\n  flex: 1;\n  height: 2.5rem;\n  border: 0;\n  background: transparent;\n  color: var(--chatbot-text);\n  font-size: 0.875rem;\n  line-height: 1.4;\n  outline: none;\n  padding: 0;\n}\n.chatbot-input:focus,\n.chatbot-input:focus-visible {\n  border: 0;\n  outline: none;\n  box-shadow: none;\n}\n.chatbot-input::placeholder {\n  color: var(--chatbot-muted);\n}\n.chatbot-send,\n.chatbot-primary-button {\n  border: 0;\n  border-radius: 9999px;\n  background: var(--chatbot-secondary);\n  color: #fff;\n  cursor: pointer;\n  font-size: 0.875rem;\n  font-weight: 700;\n  padding: 0.6rem 1rem;\n  transition: opacity 0.2s ease, background-color 0.2s ease;\n}\n.chatbot-send {\n  flex: 0 0 auto;\n  min-width: 4.5rem;\n  min-height: 2.75rem;\n  padding: 0.6rem 1.1rem;\n}\n.chatbot-send:hover,\n.chatbot-primary-button:hover {\n  background: var(--chatbot-primary);\n}\n.chatbot-send:disabled,\n.chatbot-primary-button:disabled,\n.chatbot-secondary-button:disabled {\n  cursor: not-allowed;\n  opacity: 0.45;\n}\n.chatbot-quick-actions {\n  display: flex;\n  flex-wrap: wrap;\n  gap: 0.5rem;\n}\n.chatbot-quick-action {\n  border: 1px solid var(--chatbot-border);\n  border-radius: 9999px;\n  background: var(--chatbot-bg);\n  color: var(--chatbot-text);\n  cursor: pointer;\n  font-size: 0.75rem;\n  font-weight: 700;\n  padding: 0.5rem 0.75rem;\n  transition:\n    transform 0.2s ease,\n    border-color 0.2s ease,\n    background-color 0.2s ease;\n}\n.chatbot-quick-action:hover {\n  transform: translateY(-2px);\n  border-color: color-mix(in srgb, var(--chatbot-primary) 25%, var(--chatbot-border));\n  background: color-mix(in srgb, var(--chatbot-primary) 4%, var(--chatbot-bg));\n}\n.chatbot-message-row {\n  display: flex;\n}\n.chatbot-message-row-user {\n  justify-content: flex-end;\n}\n.chatbot-message-row-assistant {\n  justify-content: flex-start;\n}\n.chatbot-message {\n  max-width: 85%;\n  border-radius: 1rem;\n  padding: 0.75rem 1rem;\n  font-size: 0.875rem;\n  line-height: 1.55;\n  box-shadow: 0 1px 2px rgba(15, 23, 42, 0.06);\n}\n.chatbot-message-user {\n  background: var(--chatbot-secondary);\n  color: #fff;\n}\n.chatbot-message-assistant {\n  border: 1px solid var(--chatbot-border);\n  background: var(--chatbot-bg);\n  color: var(--chatbot-text);\n}\n.chatbot-message-text {\n  margin: 0;\n  white-space: pre-wrap;\n}\n.chatbot-typing {\n  display: inline-flex;\n  align-items: center;\n  gap: 0.35rem;\n  min-width: 4rem;\n  min-height: 2.75rem;\n}\n.chatbot-typing span {\n  width: 0.45rem;\n  height: 0.45rem;\n  border-radius: 9999px;\n  background: color-mix(in srgb, var(--chatbot-text) 45%, transparent);\n  animation: chatbot-typing-bounce 1s ease-in-out infinite;\n}\n.chatbot-typing span:nth-child(2) {\n  animation-delay: 0.15s;\n}\n.chatbot-typing span:nth-child(3) {\n  animation-delay: 0.3s;\n}\n@keyframes chatbot-typing-bounce {\n  0%, 80%, 100% {\n    opacity: 0.35;\n    transform: translateY(0);\n  }\n  40% {\n    opacity: 1;\n    transform: translateY(-0.25rem);\n  }\n}\n.chatbot-sources {\n  display: flex;\n  flex-wrap: wrap;\n  gap: 0.5rem;\n  margin-top: 0.75rem;\n  color: var(--chatbot-muted);\n  font-size: 0.6875rem;\n}\n.chatbot-source {\n  border: 1px solid var(--chatbot-border);\n  border-radius: 9999px;\n  padding: 0.25rem 0.5rem;\n}\n.chatbot-lead-form {\n  display: flex;\n  flex-direction: column;\n  gap: 1rem;\n  border: 1px solid var(--chatbot-border);\n  border-radius: 1.5rem;\n  background: var(--chatbot-surface);\n  padding: 1rem;\n}\n.chatbot-lead-header,\n.chatbot-lead-actions {\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  gap: 1rem;\n}\n.chatbot-lead-header {\n  align-items: flex-start;\n}\n.chatbot-lead-kicker,\n.chatbot-field-label {\n  color: var(--chatbot-muted);\n}\n.chatbot-lead-title {\n  margin: 0.25rem 0 0;\n  color: var(--chatbot-text);\n  font-size: 1.125rem;\n  line-height: 1.35;\n}\n.chatbot-link-button {\n  color: var(--chatbot-muted);\n  font-size: 0.875rem;\n}\n.chatbot-link-button:hover {\n  color: var(--chatbot-text);\n}\n.chatbot-steps {\n  display: flex;\n  align-items: center;\n  gap: 0.5rem;\n  color: var(--chatbot-muted);\n  font-size: 0.6875rem;\n  font-weight: 700;\n  letter-spacing: 0.12em;\n  text-transform: uppercase;\n}\n.chatbot-step {\n  border-radius: 9999px;\n  background: color-mix(in srgb, var(--chatbot-text) 5%, transparent);\n  color: color-mix(in srgb, var(--chatbot-text) 48%, transparent);\n  padding: 0.25rem 0.5rem;\n}\n.chatbot-step-active {\n  background: var(--chatbot-secondary);\n  color: #fff;\n}\n.chatbot-field-grid {\n  display: grid;\n  gap: 0.75rem;\n}\n.chatbot-field-stack {\n  display: flex;\n  flex-direction: column;\n  gap: 0.75rem;\n}\n.chatbot-field {\n  display: flex;\n  flex-direction: column;\n  gap: 0.375rem;\n}\n.chatbot-field input,\n.chatbot-field select,\n.chatbot-field textarea,\n.input-field {\n  width: 100%;\n  border: 1px solid var(--chatbot-border);\n  border-radius: 0.875rem;\n  background: var(--chatbot-bg);\n  color: var(--chatbot-text);\n  font: inherit;\n  padding: 0.8rem 0.9rem;\n  outline: none;\n}\n.chatbot-field input:focus,\n.chatbot-field select:focus,\n.chatbot-field textarea:focus,\n.input-field:focus {\n  border-color: color-mix(in srgb, var(--chatbot-primary) 45%, var(--chatbot-border));\n  box-shadow: 0 0 0 4px rgba(99, 102, 241, 0.1);\n}\n.chatbot-textarea {\n  min-height: 7rem;\n  resize: vertical;\n}\n.chatbot-hidden {\n  display: none;\n}\n.chatbot-error {\n  border-radius: 1rem;\n  background: var(--chatbot-danger-bg);\n  color: var(--chatbot-danger);\n  font-size: 0.875rem;\n  margin: 0;\n  padding: 0.5rem 0.75rem;\n}\n.chatbot-secondary-button {\n  border: 1px solid var(--chatbot-border);\n  border-radius: 9999px;\n  background: var(--chatbot-bg);\n  color: var(--chatbot-text);\n  cursor: pointer;\n  font-size: 0.875rem;\n  font-weight: 700;\n  padding: 0.6rem 1rem;\n}\n.chatbot-scrollbar::-webkit-scrollbar {\n  width: 0.7rem;\n}\n.chatbot-scrollbar::-webkit-scrollbar-thumb {\n  border: 2px solid transparent;\n  border-radius: 9999px;\n  background: color-mix(in srgb, var(--chatbot-text) 16%, transparent);\n  background-clip: content-box;\n}\n.chatbot-scrollbar::-webkit-scrollbar-track {\n  background: transparent;\n}\n@media (min-width: 640px) {\n  .chatbot-field-grid {\n    grid-template-columns: repeat(2, minmax(0, 1fr));\n  }\n  .chatbot-field-full {\n    grid-column: span 2;\n  }\n}\n@media (max-width: 520px) {\n  .chatbot-toggle {\n    right: 1rem;\n    bottom: 1rem;\n    padding: 0.65rem;\n  }\n  .chatbot-toggle-label,\n  .chatbot-toggle-spark {\n    display: none;\n  }\n  .chatbot-window {\n    right: 0.75rem;\n    bottom: 5.25rem;\n    width: calc(100vw - 1.5rem);\n    max-height: calc(100vh - 6.5rem);\n  }\n}\n");

// src/components/ChatbotWidget.tsx
var import_react3 = require("react");

// src/components/ChatWindow.tsx
var import_react = require("react");

// src/components/MessageBubble.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function MessageBubble({ message }) {
  const isUser = message.role === "user";
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", { className: `chatbot-message-row ${isUser ? "chatbot-message-row-user" : "chatbot-message-row-assistant"}`, children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("div", { className: `chatbot-message ${isUser ? "chatbot-message-user" : "chatbot-message-assistant"}`, children: [
    /* @__PURE__ */ (0, import_jsx_runtime.jsx)("p", { className: "chatbot-message-text", children: message.content }),
    message.sources && message.sources.length > 0 && !isUser && /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", { className: "chatbot-sources", children: message.sources.map((source) => {
      var _a;
      return /* @__PURE__ */ (0, import_jsx_runtime.jsx)("span", { className: "chatbot-source", children: source.title }, `${source.title}-${(_a = source.url) != null ? _a : "source"}`);
    }) })
  ] }) });
}
var MessageBubble_default = MessageBubble;

// src/components/QuickActions.tsx
var import_jsx_runtime2 = require("react/jsx-runtime");
function QuickActions({ actions, onAction }) {
  if (!actions.length) {
    return null;
  }
  return /* @__PURE__ */ (0, import_jsx_runtime2.jsx)("div", { className: "chatbot-quick-actions", children: actions.map((action) => /* @__PURE__ */ (0, import_jsx_runtime2.jsx)("button", { type: "button", onClick: () => onAction(action), className: "chatbot-quick-action", children: action }, action)) });
}
var QuickActions_default = QuickActions;

// src/components/ChatWindow.tsx
var import_jsx_runtime3 = require("react/jsx-runtime");
function ChatWindow({
  open,
  messages,
  quickActions,
  input,
  botName,
  botEyebrow,
  inputPlaceholder,
  loading,
  onClose,
  onInputChange,
  onSubmit,
  onQuickAction,
  children
}) {
  const bodyRef = (0, import_react.useRef)(null);
  (0, import_react.useEffect)(() => {
    if (!open || !bodyRef.current) {
      return;
    }
    bodyRef.current.scrollTo({
      top: bodyRef.current.scrollHeight,
      behavior: "smooth"
    });
  }, [open, messages.length, loading, children]);
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { className: `chatbot-window ${open ? "chatbot-window-open" : "chatbot-window-closed"}`, children: [
    /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { className: "chatbot-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("div", { className: "chatbot-header-glow" }),
      /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { className: "chatbot-header-content", children: [
        /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { children: [
          /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("p", { className: "chatbot-eyebrow", children: botEyebrow }),
          /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("h2", { className: "chatbot-heading", children: botName })
        ] }),
        /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("button", { type: "button", onClick: onClose, className: "chatbot-close", children: "Close" })
      ] })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { ref: bodyRef, className: "chatbot-scrollbar chatbot-body", children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(QuickActions, { actions: quickActions, onAction: onQuickAction }),
      messages.map((message) => /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(MessageBubble, { message }, message.id)),
      loading && /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(TypingIndicator, {}),
      children
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("form", { onSubmit, className: "chatbot-form", children: /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { className: "chatbot-input-shell", children: [
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(
        "input",
        {
          name: "chat-message",
          "aria-label": "Type your message",
          value: input,
          onChange: (event) => onInputChange(event.target.value),
          placeholder: inputPlaceholder,
          className: "chatbot-input"
        }
      ),
      /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("button", { type: "submit", disabled: loading || !input.trim(), className: "chatbot-send", children: loading ? "..." : "Send" })
    ] }) })
  ] });
}
function TypingIndicator() {
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("div", { className: "chatbot-message-row chatbot-message-row-assistant", children: /* @__PURE__ */ (0, import_jsx_runtime3.jsxs)("div", { className: "chatbot-message chatbot-message-assistant chatbot-typing", "aria-live": "polite", "aria-label": "Assistant is typing", children: [
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("span", {}),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("span", {}),
    /* @__PURE__ */ (0, import_jsx_runtime3.jsx)("span", {})
  ] }) });
}
var ChatWindow_default = ChatWindow;

// src/components/LeadForm.tsx
var import_react2 = require("react");
var import_jsx_runtime4 = require("react/jsx-runtime");
var defaultLeadFormConfig = {
  serviceOptions: [
    "Web Development",
    "UI/UX Design",
    "App Development",
    "Digital Marketing",
    "Automation & Integrations",
    "Brand Identity",
    "Other"
  ],
  budgetOptions: ["Under $10,000", "$10,000 - $25,000", "$25,000 - $50,000", "$50,000+", "Not sure yet"],
  contactTimeOptions: ["Immediately", "Within 1 month", "1-3 months", "Flexible"],
  showCompanyField: true,
  showPhoneField: true,
  leadKicker: "Get in touch"
};
function LeadForm({ sourcePage, intent, config, onCancel, onSubmit }) {
  const resolvedConfig = (0, import_react2.useMemo)(() => __spreadValues(__spreadValues({}, defaultLeadFormConfig), config), [config]);
  const [step, setStep] = (0, import_react2.useState)(1);
  const [isSubmitting, setIsSubmitting] = (0, import_react2.useState)(false);
  const [error, setError] = (0, import_react2.useState)(null);
  const [formState, setFormState] = (0, import_react2.useState)({
    name: "",
    email: "",
    phone: "",
    companyName: "",
    requiredService: intent === "consultation" ? "Book a Free Consultation" : intent === "contact" ? "Contact Us" : "",
    budget: "",
    projectDetails: "",
    preferredContactTime: "",
    sourcePage,
    honeypot: "",
    submittedAt: Date.now()
  });
  const headline = (0, import_react2.useMemo)(() => {
    switch (intent) {
      case "consultation":
        return "Book a Free Consultation";
      case "contact":
        return "Contact Us";
      case "support":
        return "Talk to Human Support";
      default:
        return "Get a Quote";
    }
  }, [intent]);
  function updateField(key, value) {
    setFormState((current) => __spreadProps(__spreadValues({}, current), { [key]: value }));
  }
  function validateStepOne() {
    if (!formState.name.trim()) {
      return "Please enter your name.";
    }
    if (!formState.email.trim() || !/^\S+@\S+\.\S+$/.test(formState.email)) {
      return "Please enter a valid email address.";
    }
    if (resolvedConfig.showPhoneField && !formState.phone.trim()) {
      return "Please enter your phone number.";
    }
    return null;
  }
  function validateStepTwo() {
    if (resolvedConfig.showCompanyField && !formState.companyName.trim()) {
      return "Please enter your company name.";
    }
    if (!formState.requiredService.trim()) {
      return "Please select a required service.";
    }
    if (!formState.budget.trim()) {
      return "Please select a budget.";
    }
    if (!formState.preferredContactTime.trim()) {
      return "Please select a preferred contact time.";
    }
    return null;
  }
  function goNext() {
    setError(null);
    if (step === 1) {
      const validationError = validateStepOne();
      if (validationError) {
        setError(validationError);
        return;
      }
    }
    if (step === 2) {
      const validationError = validateStepTwo();
      if (validationError) {
        setError(validationError);
        return;
      }
    }
    setStep((current) => Math.min(current + 1, 3));
  }
  async function handleSubmit(event) {
    event.preventDefault();
    setError(null);
    if (!formState.projectDetails.trim()) {
      setError("Please describe your project.");
      return;
    }
    if (formState.honeypot.trim()) {
      setError("Submission rejected.");
      return;
    }
    setIsSubmitting(true);
    try {
      await onSubmit(__spreadProps(__spreadValues({}, formState), {
        submittedAt: Date.now()
      }));
    } catch (submissionError) {
      setError(submissionError instanceof Error ? submissionError.message : "Something went wrong.");
    } finally {
      setIsSubmitting(false);
    }
  }
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("form", { onSubmit: handleSubmit, className: "chatbot-lead-form", children: [
    /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { className: "chatbot-lead-header", children: [
      /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { children: [
        /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("p", { className: "chatbot-lead-kicker", children: resolvedConfig.leadKicker }),
        /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("h3", { className: "chatbot-lead-title", children: headline })
      ] }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { type: "button", onClick: onCancel, className: "chatbot-link-button", children: "Cancel" })
    ] }),
    /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("div", { className: "chatbot-steps", children: [1, 2, 3].map((item) => /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("span", { className: `chatbot-step ${step >= item ? "chatbot-step-active" : ""}`, children: [
      "Step ",
      item
    ] }, item)) }),
    step === 1 && /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { className: "chatbot-field-grid", children: [
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Name", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          value: formState.name,
          onChange: (event) => updateField("name", event.target.value),
          className: "input-field",
          placeholder: "Your name",
          required: true
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Email", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          type: "email",
          value: formState.email,
          onChange: (event) => updateField("email", event.target.value),
          className: "input-field",
          placeholder: "you@company.com",
          required: true
        }
      ) }),
      resolvedConfig.showPhoneField && /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Phone / WhatsApp", className: "chatbot-field-full", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          value: formState.phone,
          onChange: (event) => updateField("phone", event.target.value),
          className: "input-field",
          placeholder: "+92...",
          required: true
        }
      ) })
    ] }),
    step === 2 && /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { className: "chatbot-field-grid", children: [
      resolvedConfig.showCompanyField && /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Company name", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          value: formState.companyName,
          onChange: (event) => updateField("companyName", event.target.value),
          className: "input-field",
          placeholder: "Company or brand",
          required: true
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Required service", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(
        "select",
        {
          value: formState.requiredService,
          onChange: (event) => updateField("requiredService", event.target.value),
          className: "input-field",
          required: true,
          children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: "", children: "Select a service" }),
            resolvedConfig.serviceOptions.map((option) => /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: option, children: option }, option))
          ]
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Budget", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(
        "select",
        {
          value: formState.budget,
          onChange: (event) => updateField("budget", event.target.value),
          className: "input-field",
          required: true,
          children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: "", children: "Select a budget" }),
            resolvedConfig.budgetOptions.map((option) => /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: option, children: option }, option))
          ]
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Preferred contact time", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)(
        "select",
        {
          value: formState.preferredContactTime,
          onChange: (event) => updateField("preferredContactTime", event.target.value),
          className: "input-field",
          required: true,
          children: [
            /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: "", children: "Select one" }),
            resolvedConfig.contactTimeOptions.map((option) => /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("option", { value: option, children: option }, option))
          ]
        }
      ) })
    ] }),
    step === 3 && /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { className: "chatbot-field-stack", children: [
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Project details", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "textarea",
        {
          value: formState.projectDetails,
          onChange: (event) => updateField("projectDetails", event.target.value),
          className: "input-field chatbot-textarea",
          placeholder: "Describe your project, goals, timeline, and any special requirements.",
          required: true
        }
      ) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Field, { label: "Source page", children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("input", { value: formState.sourcePage, className: "input-field", readOnly: true }) }),
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "input",
        {
          value: formState.honeypot,
          onChange: (event) => updateField("honeypot", event.target.value),
          className: "chatbot-hidden",
          tabIndex: -1,
          autoComplete: "off",
          "aria-hidden": "true"
        }
      )
    ] }),
    error && /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("p", { className: "chatbot-error", children: error }),
    /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("div", { className: "chatbot-lead-actions", children: [
      /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
        "button",
        {
          type: "button",
          onClick: () => setStep((current) => Math.max(current - 1, 1)),
          disabled: step === 1 || isSubmitting,
          className: "chatbot-secondary-button",
          children: "Back"
        }
      ),
      step < 3 ? /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { type: "button", onClick: goNext, className: "chatbot-primary-button", children: "Next" }) : /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("button", { type: "submit", disabled: isSubmitting, className: "chatbot-primary-button", children: isSubmitting ? "Submitting..." : "Send request" })
    ] })
  ] });
}
function Field({ label, className, children }) {
  return /* @__PURE__ */ (0, import_jsx_runtime4.jsxs)("label", { className: `chatbot-field ${className != null ? className : ""}`, children: [
    /* @__PURE__ */ (0, import_jsx_runtime4.jsx)("span", { className: "chatbot-field-label", children: label }),
    children
  ] });
}
var LeadForm_default = LeadForm;

// src/lib/ragPipeline.ts
async function searchDocuments(embedding, supabaseUrl, supabaseAnonKey, matchCount, matchThreshold) {
  const response = await fetch(`${supabaseUrl.replace(/\/$/, "")}/rest/v1/rpc/match_documents`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      apikey: supabaseAnonKey,
      Authorization: `Bearer ${supabaseAnonKey}`
    },
    body: JSON.stringify({
      query_embedding: embedding,
      match_count: matchCount,
      match_threshold: matchThreshold
    })
  });
  if (!response.ok) {
    throw new Error(`Supabase vector search failed: ${response.status}`);
  }
  return await response.json();
}
var HANDOFF_PHRASES = [
  "i don't have",
  "i do not have",
  "not in the context",
  "cannot find",
  "no information",
  "i'm not sure",
  "i am not sure",
  "outside my knowledge",
  "not available in",
  "unable to find"
];
function detectHandoff(answer) {
  const lower = answer.toLowerCase();
  return HANDOFF_PHRASES.some((phrase) => lower.includes(phrase));
}
function buildSystemPrompt(chunks) {
  const context = chunks.map((chunk, index) => {
    var _a, _b;
    return `[${index + 1}] ${(_b = (_a = chunk.metadata) == null ? void 0 : _a.title) != null ? _b : "Source"}
${chunk.content}`;
  }).join("\n\n---\n\n");
  return [
    "You are a helpful assistant. Answer the user's question using ONLY the context provided below.",
    "If the context does not contain enough information to fully answer, clearly say so.",
    "Never make up or guess information that is not in the context.",
    "Be concise and direct.",
    "",
    "Context:",
    context
  ].join("\n");
}
async function runRagPipeline(question, conversation, config) {
  const {
    embeddingAdapter,
    llmAdapter,
    supabaseUrl,
    supabaseAnonKey,
    matchCount = 8,
    matchThreshold = 0.5,
    conversationWindow = 6
  } = config;
  const embedding = await embeddingAdapter(question);
  const chunks = await searchDocuments(embedding, supabaseUrl, supabaseAnonKey, matchCount, matchThreshold);
  if (chunks.length === 0) {
    return { answer: "", sources: [], needsHumanHandoff: true };
  }
  const systemPrompt = buildSystemPrompt(chunks);
  const recentConversation = conversation.slice(-conversationWindow);
  const { answer, needsHumanHandoff: adapterHandoff } = await llmAdapter({
    question,
    context: chunks.map((chunk) => chunk.content).join("\n\n"),
    conversation: recentConversation,
    systemPrompt
  });
  const needsHumanHandoff = adapterHandoff != null ? adapterHandoff : detectHandoff(answer) || !answer;
  const sources = chunks.slice(0, 8).map((chunk) => {
    var _a, _b, _c, _d;
    return {
      title: (_b = (_a = chunk.metadata) == null ? void 0 : _a.title) != null ? _b : "Source",
      url: (_d = (_c = chunk.metadata) == null ? void 0 : _c.url) != null ? _d : void 0,
      similarity: chunk.similarity
    };
  });
  return { answer: answer || "", sources, needsHumanHandoff };
}

// src/components/ChatbotWidget.tsx
var import_jsx_runtime5 = require("react/jsx-runtime");
var defaultWelcomeMsg = "Hi! I'm here to help. Ask me anything or pick one of the quick actions below.";
var defaultFallbackMsg = "I don't have enough information to answer that. Would you like to connect with our team?";
var defaultQuickActions = [
  "View Services",
  "View Case Studies",
  "Get a Quote",
  "Book a Free Consultation",
  "Contact Us",
  "Talk to Human Support"
];
var defaultBotName = "AI Assistant";
var defaultBotEyebrow = "Support";
var defaultToggleLabel = "Chat with us";
var defaultInputPlaceholder = "Type your message...";
function createInitialMessage(welcomeMsg) {
  return {
    id: crypto.randomUUID(),
    role: "assistant",
    content: welcomeMsg
  };
}
function buildConversation(messages) {
  return messages.map((message) => ({
    role: message.role,
    content: message.content
  }));
}
function ChatbotWidget({
  embeddingAdapter,
  llmAdapter,
  supabaseUrl,
  supabaseAnonKey,
  matchCount,
  matchThreshold,
  conversationWindow,
  botName = defaultBotName,
  botEyebrow = defaultBotEyebrow,
  toggleLabel = defaultToggleLabel,
  inputPlaceholder = defaultInputPlaceholder,
  welcomeMsg = defaultWelcomeMsg,
  theme = "light",
  quickActions = defaultQuickActions,
  fallbackMsg = defaultFallbackMsg,
  leadFormConfig,
  onLeadSubmit,
  leadEndpoint
}) {
  const [open, setOpen] = (0, import_react3.useState)(false);
  const [messages, setMessages] = (0, import_react3.useState)(() => [createInitialMessage(welcomeMsg)]);
  const [input, setInput] = (0, import_react3.useState)("");
  const [loading, setLoading] = (0, import_react3.useState)(false);
  const [leadFormOpen, setLeadFormOpen] = (0, import_react3.useState)(false);
  (0, import_react3.useEffect)(() => {
    setMessages([createInitialMessage(welcomeMsg)]);
  }, [welcomeMsg]);
  const quickActionList = (0, import_react3.useMemo)(() => quickActions.filter((action) => action.trim().length > 0), [quickActions]);
  const resolvedLeadFormConfig = (0, import_react3.useMemo)(() => __spreadValues(__spreadValues({}, defaultLeadFormConfig), leadFormConfig), [leadFormConfig]);
  function getSourcePage() {
    return typeof window !== "undefined" ? window.location.pathname : "/";
  }
  async function submitQuestion(question) {
    const trimmedQuestion = question.trim();
    if (!trimmedQuestion || loading) {
      return;
    }
    const nextUserMessage = {
      id: crypto.randomUUID(),
      role: "user",
      content: trimmedQuestion
    };
    setMessages((current) => [...current, nextUserMessage]);
    setInput("");
    setLoading(true);
    try {
      const result = await runRagPipeline(trimmedQuestion, buildConversation([...messages, nextUserMessage]), {
        embeddingAdapter,
        llmAdapter,
        supabaseUrl,
        supabaseAnonKey,
        matchCount,
        matchThreshold,
        conversationWindow
      });
      const assistantMessage = {
        id: crypto.randomUUID(),
        role: "assistant",
        content: result.answer || fallbackMsg,
        needsHumanHandoff: result.needsHumanHandoff,
        sources: result.sources
      };
      setMessages((current) => [...current, assistantMessage]);
      if (assistantMessage.needsHumanHandoff) {
        setLeadFormOpen(true);
      }
    } catch (e) {
      setMessages((current) => [
        ...current,
        {
          id: crypto.randomUUID(),
          role: "assistant",
          content: fallbackMsg,
          needsHumanHandoff: true
        }
      ]);
      setLeadFormOpen(true);
    } finally {
      setLoading(false);
    }
  }
  function handleQuickAction(action) {
    void submitQuestion(action);
  }
  async function submitLead(values) {
    if (leadEndpoint) {
      const response = await fetch(leadEndpoint, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(values)
      });
      if (!response.ok) {
        throw new Error("Unable to submit lead.");
      }
    } else if (onLeadSubmit) {
      await onLeadSubmit(values);
    }
    setLeadFormOpen(false);
    setMessages((current) => [
      ...current,
      {
        id: crypto.randomUUID(),
        role: "assistant",
        content: "Thanks! Your details have been received. Someone will be in touch with you shortly."
      }
    ]);
  }
  const leadForm = leadFormOpen ? /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
    LeadForm,
    {
      sourcePage: getSourcePage(),
      config: resolvedLeadFormConfig,
      onCancel: () => setLeadFormOpen(false),
      onSubmit: submitLead
    }
  ) : null;
  return /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)("div", { className: `chatbot-widget-shell chatbot-theme-${theme}`, children: [
    /* @__PURE__ */ (0, import_jsx_runtime5.jsxs)(
      "button",
      {
        type: "button",
        onClick: () => setOpen((current) => !current),
        className: "chatbot-toggle",
        "aria-label": open ? "Close chat" : "Open chat",
        "aria-expanded": open,
        children: [
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("span", { className: "chatbot-toggle-icon", "aria-hidden": "true", children: /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("span", { className: "chatbot-toggle-dot" }) }),
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("span", { className: "chatbot-toggle-label", children: toggleLabel }),
          /* @__PURE__ */ (0, import_jsx_runtime5.jsx)("span", { className: "chatbot-toggle-spark", "aria-hidden": "true", children: "\u2726" })
        ]
      }
    ),
    /* @__PURE__ */ (0, import_jsx_runtime5.jsx)(
      ChatWindow,
      {
        open,
        messages,
        quickActions: quickActionList,
        input,
        botName,
        botEyebrow,
        inputPlaceholder,
        loading,
        onClose: () => setOpen(false),
        onInputChange: setInput,
        onSubmit: (event) => {
          event.preventDefault();
          void submitQuestion(input);
        },
        onQuickAction: handleQuickAction,
        children: leadForm
      }
    )
  ] });
}

// src/lib/adapterFactories.ts
function createGeminiEmbeddingAdapter(apiKey) {
  return async (text) => {
    const response = await fetch(
      `https://generativelanguage.googleapis.com/v1beta/models/gemini-embedding-001:embedContent?key=${apiKey}`,
      {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          model: "models/gemini-embedding-001",
          content: { parts: [{ text }] }
        })
      }
    );
    if (!response.ok) throw new Error(`Gemini embedding failed: ${response.status}`);
    const data = await response.json();
    return data.embedding.values;
  };
}
function createOpenAIEmbeddingAdapter(apiKey, model = "text-embedding-3-small") {
  return async (text) => {
    const response = await fetch("https://api.openai.com/v1/embeddings", {
      method: "POST",
      headers: { "Content-Type": "application/json", Authorization: `Bearer ${apiKey}` },
      body: JSON.stringify({ model, input: text })
    });
    if (!response.ok) throw new Error(`OpenAI embedding failed: ${response.status}`);
    const data = await response.json();
    return data.data[0].embedding;
  };
}
function createCohereEmbeddingAdapter(apiKey, model = "embed-english-v3.0") {
  return async (text) => {
    const response = await fetch("https://api.cohere.com/v2/embed", {
      method: "POST",
      headers: { "Content-Type": "application/json", Authorization: `Bearer ${apiKey}` },
      body: JSON.stringify({
        model,
        texts: [text],
        input_type: "search_query",
        embedding_types: ["float"]
      })
    });
    if (!response.ok) throw new Error(`Cohere embedding failed: ${response.status}`);
    const data = await response.json();
    return data.embeddings.float[0];
  };
}
function createOpenAICompatibleLLMAdapter(config) {
  const { apiKey, baseUrl, model, temperature = 0.3, maxTokens = 1024 } = config;
  return async ({ systemPrompt, question, conversation }) => {
    var _a, _b, _c;
    const response = await fetch(`${baseUrl.replace(/\/$/, "")}/v1/chat/completions`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${apiKey}`
      },
      body: JSON.stringify({
        model,
        temperature,
        max_tokens: maxTokens,
        messages: [
          { role: "system", content: systemPrompt },
          ...conversation,
          { role: "user", content: question }
        ]
      })
    });
    if (!response.ok) throw new Error(`LLM request failed: ${response.status}`);
    const data = await response.json();
    return { answer: ((_c = (_b = (_a = data.choices[0]) == null ? void 0 : _a.message) == null ? void 0 : _b.content) != null ? _c : "").trim() };
  };
}
var createDeepSeekLLMAdapter = (apiKey, model = "deepseek-chat") => createOpenAICompatibleLLMAdapter({ apiKey, baseUrl: "https://api.deepseek.com", model });
var createOpenAILLMAdapter = (apiKey, model = "gpt-4o-mini") => createOpenAICompatibleLLMAdapter({ apiKey, baseUrl: "https://api.openai.com", model });
var createGroqLLMAdapter = (apiKey, model = "llama3-8b-8192") => createOpenAICompatibleLLMAdapter({ apiKey, baseUrl: "https://api.groq.com/openai", model });
var createMistralLLMAdapter = (apiKey, model = "mistral-small-latest") => createOpenAICompatibleLLMAdapter({ apiKey, baseUrl: "https://api.mistral.ai", model });
var createTogetherLLMAdapter = (apiKey, model) => createOpenAICompatibleLLMAdapter({ apiKey, baseUrl: "https://api.together.xyz", model });
var createOllamaLLMAdapter = (model, baseUrl = "http://localhost:11434") => createOpenAICompatibleLLMAdapter({ apiKey: "ollama", baseUrl, model });
function createAnthropicLLMAdapter(apiKey, model = "claude-3-5-haiku-20241022") {
  return async ({ systemPrompt, question, conversation }) => {
    var _a, _b;
    const response = await fetch("https://api.anthropic.com/v1/messages", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "x-api-key": apiKey,
        "anthropic-version": "2023-06-01"
      },
      body: JSON.stringify({
        model,
        max_tokens: 1024,
        system: systemPrompt,
        messages: [...conversation.map((turn) => ({ role: turn.role, content: turn.content })), { role: "user", content: question }]
      })
    });
    if (!response.ok) throw new Error(`Anthropic request failed: ${response.status}`);
    const data = await response.json();
    return { answer: ((_b = (_a = data.content[0]) == null ? void 0 : _a.text) != null ? _b : "").trim() };
  };
}
function createGeminiLLMAdapter(apiKey, model = "gemini-2.5-flash") {
  return async ({ systemPrompt, question, conversation }) => {
    var _a, _b, _c, _d;
    const contents = [
      ...conversation.map((turn) => ({
        role: turn.role === "assistant" ? "model" : "user",
        parts: [{ text: turn.content }]
      })),
      { role: "user", parts: [{ text: question }] }
    ];
    const response = await fetch(
      `https://generativelanguage.googleapis.com/v1beta/models/${model}:generateContent?key=${apiKey}`,
      {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          system_instruction: { parts: [{ text: systemPrompt }] },
          contents
        })
      }
    );
    if (!response.ok) throw new Error(`Gemini LLM request failed: ${response.status}`);
    const data = await response.json();
    return { answer: ((_d = (_c = (_b = (_a = data.candidates[0]) == null ? void 0 : _a.content) == null ? void 0 : _b.parts[0]) == null ? void 0 : _c.text) != null ? _d : "").trim() };
  };
}

// src/lib/supabaseSetup.ts
var SUPABASE_SETUP_SQL = `
-- Enable pgvector
create extension if not exists vector;

-- Documents table (change vector(3072) to match your embedding model)
create table if not exists documents (
  id          uuid primary key default gen_random_uuid(),
  content     text not null,
  embedding   vector(3072),
  metadata    jsonb default '{}'::jsonb,
  created_at  timestamptz default now()
);

-- HNSW index on halfvec cast (required for dims > 2000; IVFFlat max is 2000)
create index if not exists documents_embedding_idx
  on documents using hnsw ((embedding::halfvec(3072)) halfvec_cosine_ops);

-- Similarity search function (change vector(3072) to match your model)
create or replace function match_documents(
  query_embedding  vector(3072),
  match_count      int   default 8,
  match_threshold  float default 0.5
)
returns table (
  id         uuid,
  content    text,
  metadata   jsonb,
  similarity float
)
language sql stable as $$
  select
    id,
    content,
    metadata,
    1 - (embedding <=> query_embedding) as similarity
  from documents
  where 1 - (embedding <=> query_embedding) > match_threshold
  order by embedding <=> query_embedding
  limit match_count;
$$;

-- Permissions: the chatbot widget uses the anon key in the browser
grant usage on schema public to anon, authenticated;
grant select on public.documents to anon, authenticated;
grant execute on function public.match_documents(vector(3072), int, float) to anon, authenticated;

alter table public.documents enable row level security;

drop policy if exists "Allow public read for RAG" on public.documents;
create policy "Allow public read for RAG"
  on public.documents
  for select
  to anon, authenticated
  using (true);
`;
async function ingestDocument(content, metadata, embedText, supabaseUrl, supabaseAnonKey) {
  const embedding = await embedText(content);
  const response = await fetch(`${supabaseUrl.replace(/\/$/, "")}/rest/v1/documents`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      apikey: supabaseAnonKey,
      Authorization: `Bearer ${supabaseAnonKey}`,
      Prefer: "return=minimal"
    },
    body: JSON.stringify({ content, embedding, metadata })
  });
  if (!response.ok) {
    throw new Error(`Failed to ingest document: ${response.status} ${await response.text()}`);
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ChatWindow,
  ChatbotWidget,
  LeadForm,
  MessageBubble,
  QuickActions,
  SUPABASE_SETUP_SQL,
  createAnthropicLLMAdapter,
  createCohereEmbeddingAdapter,
  createDeepSeekLLMAdapter,
  createGeminiEmbeddingAdapter,
  createGeminiLLMAdapter,
  createGroqLLMAdapter,
  createMistralLLMAdapter,
  createOllamaLLMAdapter,
  createOpenAICompatibleLLMAdapter,
  createOpenAIEmbeddingAdapter,
  createOpenAILLMAdapter,
  createTogetherLLMAdapter,
  ingestDocument,
  runRagPipeline
});
